components/navbar/ProfileDropdown.vue

```diff
@@ -289,11 +292,17 @@ export default class ProfileDropdown extends mixins(
   }
 
   get account() {
-    return this.$store.getters.getAuthAddress
+    if (this.identityStore.getAuthAddress) {
+      console.log('fetching balance...')
+      this.identityStore.fetchBalance({
+        address: this.identityStore.getAuthAddress,
+      })
```

Comment:
We should not fetch balance every time execute the `get` function. IMO it should be wrapped in the account watch hook. Or put it into `set` account function.

Reply:
> I use it to fetchBalance on app load, any recommendation or maybe I can just
> 
> ```ts
> onMounted(() => {
>   if (this.auth.address) {
>     this.fetchBalance({ address: this.auth.address })
>   }
> }
> ```
> 
> in identity store

Let's take a try. IMO, we could try to do the first call on the mounted hook in the wallet asset component. I'm not sure if we could put it into the identity store.

Reply:
Oh, I see there is already a commit for it.